site/components/map/Map.tsx

```diff
@@ -261,14 +361,137 @@ const Map = ({
+                        type="geojson"
+                        data={createLineString(searchedRoute.inbound, searchedRoute?.serviceId)}
+                    >
+                        <Layer
+                            id={`services-inbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-inbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyle}
+                        />
+                        <Layer
+                            id={`services-highlighted-inbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-inbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyleHighlight}
+                            filter={filter}
+                        />
+                    </Source>
+                ) : null,
+            ),
+
+        [filter],
+    );
+
+    const getSourcesOutbound = useCallback(
+        (searchedRoutes: Partial<(Routes & { serviceId: number })[]>) =>
+            searchedRoutes.map((searchedRoute) =>
+                searchedRoute?.outbound ? (
+                    <Source
+                        key={searchedRoute?.serviceId}
+                        id={`outbound-route-${searchedRoute?.serviceId}`}
+                        type="geojson"
+                        data={createLineString(searchedRoute.outbound, searchedRoute?.serviceId)}
+                    >
+                        <Layer
+                            id={`services-outbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-outbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyle}
+                        />
+                        <Layer
+                            id={`services-highlighted-outbound-${searchedRoute?.serviceId}`}
+                            type="line"
+                            source={`services-outbound-${searchedRoute?.serviceId}`}
+                            layout={lineLayout}
+                            paint={lineStyleHighlight}
+                            filter={filter}
+                        />
+                    </Source>
+                ) : null,
+            ),
+        [filter],
+    );
+
+    const getInteractiveLayerIds = useCallback(
+        () =>
+            searchedRoutes && searchedRoutes.length > 0
+                ? searchedRoutes.flatMap((sr) => {
+                      if (sr?.inbound && sr.outbound) {
+                          return [
+                              `services-inbound-${sr?.serviceId || ""}`,
+                              `services-outbound-${sr?.serviceId || ""}`,
+                          ];
+                      } else if (sr?.inbound) {
```

Comment:
I dont think we need the else's because we have returns in each if.

Reply:
done